src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -1,22 +1,53 @@
 package ma.hibernate.dao;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import lombok.extern.log4j.Log4j;
 import ma.hibernate.model.Phone;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
+@Log4j
 public class PhoneDaoImpl extends AbstractDao implements PhoneDao {
     public PhoneDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.persist(phone);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Couldn't insert phone entity " + phone.toString(), e);
+        }
+        return phone;
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            List<Predicate> predicates = new ArrayList<>();
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                predicates.add(root.get(entry.getKey()).in(Arrays.asList(entry.getValue())));
+            }
+            query.select(root).where(predicates.toArray(new Predicate[]{}));
+            return session.createQuery(query).getResultList();
+        }
```

Comment:
Why we're not using catch block here?

Reply:
I am not sure is it ok for this situation, but as Bohdan says, when it is select query we do not need catch and rollback.

Reply:
we use no transaction while read operation, so there is nothing to rollback. With catch it's up to you, if you wanna to add some customer message and rethrow exception you can do it, but it's not obligatory